simpleui-server/src/simpleui-server.ts

```diff
@@ -88,14 +88,40 @@ export class SimpleUIServer {
                     const resourceEndIndex = link.url.indexOf('&', resourceIndex);
                     link.url = link.url.substring(0, resourceIndex + 9) + cmdVars.urlResource + link.url.substring(resourceEndIndex);
                 }
-            })
+            });
+        }
+        if (cmdVars.readonly) {
+            Logger.log(LogLevel.INFO, "Running app in readonly mode");
+            props['readonly'] = true;
+            // append 'Read Only ' to title of app
+            if (props['instance']['name'].substring(0,9) != 'Read Only') {
+                props['instance']['name'] = `Read Only ${props['instance']['name']}`;
+            }
+            // append '(RO) ' to beginning of tab title
+            for (let i = 0; i < props['tab'].length; i++) {
+                if (props['tab'][i]['name'].substring(0,4) != '(RO)') {
+                    props['tab'][i]['name'] = `(RO) ${props['tab'][i]['name']}`;
+                }
+
+                if (props['tab'][i]['disabled_buttons'] != 'true') {
+                    props['tab'][i]['disabled_buttons'] = 'true';
+                }
+            }
+            // remove params applink if present
+            props.appLink.forEach(p => {
+                if (p.url.includes('Parameters.html')) {
+                    p.url = '';
+                    p.name = '';
+                    p.tooltip = '';
+                }
+            });
         }
 
         return props;
     }
 
     static parseCommandLine(cmdLine: string): any {
-        const cmdVars: CommandArgs = <CommandArgs> {
+        const cmdVars: CommandArgs = {
```

Comment:
Not that type the code is all that type-safe generally, but did we lose a tiny bit by removing `<CommandArgs>`.  I.e., should the `CommandArgs` type need be updated (or removed)?

Could be that I'm misreading typescript.

Reply:
The reason the `CommandArgs` cast was needed was because the `examples` property was not declared on the `CommandArgs` interface. After adding the `examples` property, we could just declare `cmdVars` as type `CommandVars`. Having the type declared and casting the same type was redundant and it seemed like having the cast there was kinda a bandaid solution to the actual problem (the interface not having the `examples` property).


In the future, Id like to cleanup the CommandVars thing but I might just wait until we re-write simpleUI